### PR TITLE
fix: Override getScreenshotAs to support the legacy base64 encoding

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
## Change list

Addresses https://github.com/SeleniumHQ/selenium/issues/11168
 
## Types of changes

- [ ] No changes in production code.
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Review by SrinivasanTarget — APPROVED** [review]

File: src/main/java/io/appium/java_client/AppiumDriver.java [hunk 1]
```diff
@@ -270,4 +277,48 @@ public Response execute(String driverCommand, Map<String, ?> parameters) {
...
+        if (outputType == OutputType.BYTES) {
+            return (X) screenshotRawBytes;
+        } else if (outputType == OutputType.BASE64) {
+            //noinspection unchecked
+            return (X) Base64.getEncoder().encodeToString(screenshotRawBytes);
+        } else if (outputType == OutputType.FILE) {
+            //noinspection unchecked
+            return (X) saveToFile(screenshotRawBytes);
+        }
+        throw new IllegalArgumentException(String.format("Unsupported %s instance", OutputType.class.getName()));
```
Comment (valfirst) [inline src/main/java/io/appium/java_client/AppiumDriver.java:322]: what about making it simpler?:

```suggestion
        return super.getScreenshotAs(new OutputType<>() {
            @Override
            public X convertFromBase64Png(String base64Png) {
            // TODO: Eventually we should not override this method.
            // TODO: Although, we have a legacy burden,
            // TODO: so it's impossible to do it the other way as of Oct 29 2022.
            // TODO: See https://github.com/SeleniumHQ/selenium/issues/11168
                return outputType.convertFromBase64Png(base64Png.replaceAll("\r?\n", ""));
            }

            @Override
            public X convertFromPngBytes(byte[] png) {
                return outputType.convertFromPngBytes(png);
            }
        });
```

Comment (mykola-mokhnach) [reply src/main/java/io/appium/java_client/AppiumDriver.java]: good idea. Changed

**Review by valfirst — APPROVED** [review]

**Comment by krishtoautomate** [comment]
Possible to create maven build of appium java client with this patch?

**Comment by sergey-lysenko** [comment]
This only fixed fullscreen screentshots, while calling method of WebElement falls back to default implementation and fails the test.

I guess the only option now (untill overriding is the only option) is to create a routine in tests which will make a full screenshot, then crop needed part based on WebElement's dimentions.

**Comment by mykola-mokhnach** [comment]
> This only fixed fullscreen screentshots, while calling method of WebElement falls back to default implementation and fails the test.
> 
> I guess the only option now (untill overriding is the only option) is to create a routine in tests which will make a full screenshot, then crop needed part based on WebElement's dimentions.

This patch is ugly and it anyway would make sense to apply it on Selenium side instead. Or simply use the newest version of an appium driver, where it generates a compatible base64-encoding.

**Pull request merged**